src/pages/TokenDetails/index.tsx

```diff
@@ -94,12 +104,14 @@ export default function TokenDetails() {
             <BreadcrumbNavLink to={`/tokens/${chainName}`}>
               <ArrowLeft size={14} /> Tokens
             </BreadcrumbNavLink>
-            <ChartSection
-              token={tokenQueryData}
-              currency={token}
-              nativeCurrency={isNative ? nativeCurrency : undefined}
-              prices={prices}
-            />
+            <Suspense fallback={<LoadingChart />}>
```

Comment:
This should be a part of `ChartSection`, so that any loading ux is a part of the component itself.

Reply:
A suspense wrapper must be a parent component, rather a wrapper within the component itself, for it to catch any suspense that is thrown

Reply:
You could use the preloaded prices query from within the `ChartSection` so that it can still catch the thrown suspense.
This should be fine - I don't think prices are used elsewhere. It's also an idiomatic usage of using a preloaded query from children components, with the parent component doing the actual fetch.